src/ott/datasets.py

```diff
@@ -105,6 +110,168 @@ def _create_sample_generators(self) -> Iterator[jnp.array]:
+      - ``swiss`` - swiss roll Sklearn dataset
+
+    init_rng: initial PRNG key
+    dim_data: data dimensionality
+    theta_rotation: angle to define the rotation matrix to rotate samples
+    offset: offset added to the Sklearn samples
+    scale: scaling factor to scale the Sklearn samples
+    std_noise: standard deviation of the gaussian additive noise
+    batch_size: batch size of the samples
+  """
+
+  name: Literal["moon", "s_curve", "swiss"]
+  init_rng: jax.Array
+  dim_data: int = 2
+  theta_rotation: float = 0.0
+  offset: Optional[jnp.ndarray] = None
+  scale: float = 1.0
+  std_noise: float = 0.01
+  batch_size: int = 1024
+
+  def __iter__(self) -> Iterator[jnp.ndarray]:
+    """Random sample generator from a Sklearn distribution.
+
+    Returns:
+    A generator of samples from the Sklearn distribution.
+    """
+    return self._create_sample_generators()
+
+  def __post_init__(self):
+
+    # check dimension consistency with dataset
+    if self.dim_data > 2:
+      assert self.name != "moon", (
+          "Moon disribution only supported in dimension 2."
+      )
+      assert self.theta_rotation == 0., (
+          "Samples rotation only supported in dimension 2."
+      )
+
+    # define offset
+    self._offset = jnp.zeros(
+        self.dim_data
+    ) if self.offset is None else self.offset
+
+    # define rotation matrix to rotate samples
+    if self.theta_rotation == 0.:
+      self.rotation_fn = lambda x: x
+    else:
+      rotation_marix = jnp.array([
+          [jnp.cos(self.theta_rotation), -jnp.sin(self.theta_rotation)],
+          [jnp.sin(self.theta_rotation),
+           jnp.cos(self.theta_rotation)],
+      ])
+      self.rotation_fn = lambda x: jnp.matmul(rotation_marix, x)
+
+  def _create_sample_generators(self) -> Iterator[jnp.ndarray]:
+    rng = jax.random.PRNGKey(0) if self.init_rng is None else self.init_rng
+
+    while True:
+      rng, _ = jax.random.split(rng)
+      seed = jax.random.randint(rng, [], minval=0, maxval=1e5).item()
+      if self.name == "moon":
+        samples, _ = sklearn.datasets.make_moons(
+            n_samples=(self.batch_size, 0),
+            random_state=seed,
+            noise=self.std_noise,
+        )
+      elif self.name == "s_curve":
+        x, _ = sklearn.datasets.make_s_curve(
+            n_samples=self.batch_size,
+            random_state=seed,
+            noise=self.std_noise,
+        )
+        samples = x[:, [2, 0]] if self.dim_data == 2 else x
+      elif self.name == "swiss":
+        x, _ = sklearn.datasets.make_swiss_roll(
+            n_samples=self.batch_size,
+            random_state=seed,
+            noise=self.std_noise,
+        )
+        samples = x[:, [2, 0]] if self.dim_data == 2 else x
+      else:
+        raise NotImplementedError(
+            f"SklearnDistribution `{self.name}` not implemented."
+        )
+
+      samples = jnp.asarray(samples, dtype=jnp.float32)
+      samples = jax.vmap(self.rotation_fn)(samples)
```

Comment:
Think you can remove the vmap here and add it before. Also, is the `vmap` even necessary? I assume `sampled` are `(n, d)` and the rotation matrix is `(d, d)` where `d=2`, no?